cmake/PostgreSQLExtension.cmake

```diff
@@ -318,16 +318,30 @@ if [ -d \"$_dir/${NAME}\" ]; then
   _dir=\"$_dir/${NAME}\"
 fi
 # Create file (using $$ for pid to avoid race conditions)
+
 for f in $(ls \"$_dir/\"*.sql | sort -V)
   do
+  filename=$(basename \"$f\")
+  if [[ $filename =~ ([0-9]+)_ ]]; then
+    migration_number=\${BASH_REMATCH[1]}
+    if grep -q \" $migration_number \" <<< \"$unique_versions\"; then
+      echo \"Error: Duplicate migration number found: $migration_number in filename: $filename\" >&2
+      exit 1
```

Comment:
@yrashk While the logic works and it throws error code 1 from the generated scripts (`script_omni_python` for example), this is not error-ing out the `make` command.

Can you please guide me on how to achieve that? Thanks!

Reply:
I apologize for forgetting about this, slipped through the cracks.

Which `make` target does it not fail on? `make psql_`, or tests, I presume?

My intuition here suggests that it has to do with how other scripts handle failure of `script_EXT`.

What I can note here:

* `script_` itself simply skips dependant `script_` invocations when they fail: https://github.com/omnigres/omnigres/blob/master/cmake/PostgreSQLExtension.cmake#L325
* pg_regress test harness doesn't check for the errors: https://github.com/omnigres/omnigres/blob/master/cmake/PostgreSQLExtension.cmake#L419
* `psql_` doesn't either: https://github.com/omnigres/omnigres/blob/master/cmake/PostgreSQLExtension.cmake#L484
* pg_yregress harness _should_ work but worth investigating: https://github.com/omnigres/omnigres/blob/master/cmake/PostgreSQLExtension.cmake#L122